src/js/fable-core/ListClass.ts

```diff
@@ -16,12 +16,26 @@ export function ofArray<T>(args: T[], base?: List<T>) {
 }
 
 export default class List<T> implements IEquatable<List<T>>, IComparable<List<T>>, Iterable<T> {
-  public head: T;
-  public tail: List<T>;
+  private _head: T;
+  private _tail: List<T>;
 
   constructor(head?: T, tail?: List<T>) {
-    this.head = head;
-    this.tail = tail;
+    this._head = head;
+    this._tail = tail;
+  }
+
+  get head(): T {
+    if (this._head != null)
```

Comment:
Will this work? what if I had a list of null?

Reply:
Good catch, `[null].Head` fails.

Reply:
Indeed this can be an issue, even `undefined` can be a problem if we change the representation of `()` and you have a `unit list`. In theory, you should be able to check `this._tail` here as well. Empty list will always have null tails.